pg_metadata/install/sql/pgmetadata/10_FUNCTION.sql

```diff
@@ -16,6 +16,41 @@ SET xmloption = content;
 SET client_min_messages = warning;
 SET row_security = off;
 
+-- calculate_fields_from_data()
+CREATE FUNCTION pgmetadata.calculate_fields_from_data() RETURNS trigger
+    LANGUAGE plpgsql
+    AS $$
+DECLARE
+    my_table text;
+    test_geom boolean;
+BEGIN
+	my_table = CONCAT(NEW.schema_name, '.', NEW.table_name);
+	EXECUTE 'SELECT COUNT(*) FROM ' || my_table
+	INTO NEW.feature_count;
+	EXECUTE 'SELECT true
+    FROM information_schema.columns 
+    WHERE table_schema=''' || NEW.schema_name || ''' AND table_name=''' || NEW.table_name || ''' and column_name=''geom'' ' INTO test_geom;
```

Comment:
To test if the table has a geometry column, I would use a direct SQL query on the `public.geometry_columns` table such as

```sql
SELECT * 
FROM geometry_columns WHERE f_table_schema = 'NOM_SCHEMA' AND f_table_name = 'NOM_TABLE' 
INTO geometry_columns_rec;
```
(this is an example, replace with the NEW.schema_name and table_name)
which will return nothing if no line is found, but a record with geometry_columns data if sucess.

If you modify the  `DECLARE` to add a new param: `geometry_columns_rec record;` which will receive the result of this query

This way you will be able to use directly the `geometry_columns_rec.geometry_type` like

```sql
NEW.geometry_type = geometry_columns_rec.geometry_type;
```